posts/2023-02-20-liberty-tools-eclipse-deep-dive.adoc

```diff
@@ -0,0 +1,118 @@
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-run-config.gif[Run Configurations,width=70%,align="center"]
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-terminal-start.png[Terminal Start,width=70%,align="center"]
+
+NOTE: The `Start in container` action will link:https://github.com/OpenLiberty/ci.maven/blob/main/docs/dev.md#devc-container-mode[start dev mode in a local Docker container]. 
+
+=== Running tests
+
+Tests are important for any application. Liberty Tools makes it easy to run your unit and integration tests by either selecting the `Run tests` action, or simply clicking `Enter` in the terminal. You can even view the results of your latest run by selecting `View test reports`.
+
+=== Debugging your app
+
+The Liberty runtime is configured with port 7777 for debugging by default. However, this requires that port to be available on your system and for you to manually attach a debugger. With Liberty Tools, you can start your app using the "Debug" action which will find an available port and automatically attach the Eclipse debugger to it. 
+
+== Get coding!
+
+=== Jakarta EE 
+
+Code completion for Jakarta EE makes it easy to add classes and methods to your app. Just type `CTRL` + `SPACE` from within any Java file to see a list of available code snippets to choose from. Additionally, diagnostics flag errors and suggest quick-fixes to implement on the spot.
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-jakarta-snippet-2.gif[Jakarta code completion,width=70%,align="center"]
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-jakarta-quick-fix-2.gif[Jakarta quick fix,width=70%,align="center"]
+
+=== MicroProfile
+
+Similar code completion is available for MicroProfile as well as type-ahead suggestions and hover descriptions for MicroProfile config properties. You can easily see what values are set for injected properties directly from your code and get lists of available variables and values to add to your configuration.
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-mp-props-hover.png[MicroProfile config property hover,width=70%,align="center"]
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-mp-props.gif[MicroProfile code completion,width=70%,align="center"]
+
+=== Liberty config editing 
+
+Liberty Tools streamlines the process of configuring the Liberty runtime through config completion for `server.xml`, `bootstrap.properties` and `server.env` files. Type `CTRL` + `SPACE` from within these files to get lists of suggested configuration elements, properties, and values to add. 
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-server-config.gif[Server config code completion,width=70%,align="center"]
+
+[.img_border_light]
+image::/img/blog/liberty-tools-eclipse-bootstrap-props.gif[Bootstrap properties code completion,width=70%,align="center"]
+
+== Learn more 
```

Comment:
Is this supposed to be empty?

Reply:
No!   Thanks for the catch.  Changing this to a "Feedback" section